docs/subinterfaces.rst

```diff
@@ -0,0 +1,45 @@
+Configuring Sub-interfaces Using Python Client for eAPI
+======================================================
+
+Sub-interfaces can be configured on Ethernet and Port-Channel interfaces. To do this in
+eAPI simply call create or delete with your sub-interface name.
+
+import pyeapi
+node = pyeapi.connect_to('veos01')
+node.api('interfaces').create('Ethernet1.1')
+
+At this point the below should be in your running configuration.
+
+!
+interface Ethernet1
+!
+interface Ethernet1.1
+!
+
+Sub-interfaces require that the primary interface be in routed mode.
+
+node.api('ipinterfaces').create('Ethernet1')
```

Comment:
I would put this as the first step since our documentation says an intf must be an L3 intf before adding a sub-interface. Ie
`node.api('ipinterfaces').create('Ethernet1')`
`node.api('interfaces').create('Ethernet1.1')`
`node.api('interfaces').set_encapsulation('Ethernet1.1', 4)`

Up to you though.